jina/drivers/rank/__init__.py

```diff
@@ -82,29 +76,36 @@ def _apply_all(self, docs: 'DocumentSet', *args, **kwargs) -> None:
             )
 
             matches = doc.matches
-            old_match_scores = {match.id: match.score.value for match in matches}
-            match_meta = (
-                {match.id: match.get_attrs(*self._exec_match_keys) for match in matches}
-                if self._exec_match_keys
-                else None
-            )
+            num_matches = len(matches)
+            old_match_scores = []
+            needs_match_meta = self._exec_match_keys is not None
+            match_meta = [] if needs_match_meta else None
+            for match in matches:
+                old_match_scores.append(match.score.value)
+                if needs_match_meta:
+                    match_meta.append(match.get_attrs(*self._exec_match_keys))
 
             # if there are no matches, no need to sort them
             if not old_match_scores:
                 continue
 
-            new_match_scores = self.exec_fn(query_meta, old_match_scores, match_meta)
-            self._sort_matches_in_place(doc, new_match_scores)
+            new_scores = self.exec_fn(old_match_scores, query_meta, match_meta)
```

Comment:
why change the name from `new_match_scores` to `new_scores` here?
We have `old_match_scores` would it be better to use `new_match_scores`?

Reply:
yes, we could call it back to nee